### PR TITLE
API design

### PR DESCRIPTION
**Pull request by kuppulur**
This PR is to enable easy review of the library API design. Once approved, we can just delete the branch to keep the repository clean. Please review and suggest any additional things I  may have missed.

### CODE REVIEW

File: api/api.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import pickle
+from typing import List
+from .exceptions import InvalidModelError
+
+from sentence_transformers import SentenceTransformer
+
+
+class Ner4Opt(object):
+
+    CLASSICAL = 'classical'
```
Comment (skadio) [inline api/api.py:10]: On one hand, I like that it matches the paper.
On the other hand, another option could be Lexical, Lexical+, Semantic, Hybrid?

File: api/api.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            self._model = pickle.load(open('models/classical.pkl', 'rb'))
+
+        elif model_name == Ner4Opt.CLASSICAL_PLUS:
+            self._model = pickle.load(open('models/classical_plus.pkl', 'rb'))
+
+        elif model_name == Ner4Opt.DEEP:
+            self._model = SentenceTransformer("skadio/optimization_finetuned_roberta")
+
+        elif model_name == Ner4Opt.HYBRID:
+            self._model = pickle.load(open('models/hybrid_crf_model.pkl', 'rb'))
```
Comment (skadio) [inline api/api.py:30]: curious how they hbyrid will access the SentenceTransfomer from opt_tuned_roberta above

**Review by skadio — APPROVED** [review]
LGTM!

File: api/api.py (hunk 2 above)
Comment (kuppulur) [reply api/api.py:30]: Great point, I will make sure there wouldn't be any duplicate instantiations.

**Review by skadio — CHANGES_REQUESTED** [review]
IF this branch is not needed anymore, please NUKE

**Pull request closed without merging**